### PR TITLE
New statistics page

### PR DESCRIPTION
**Pull request by vicentemendozam**
Where
=====
* **Related Issue:** #281 

What
====
- New statistics page for Agendas

How
===
- Add tables with new statisctis

Screenshots
===========
issue_281

Test
====
- Routing test for the new page

Deployment
==========
- Migration

### CODE REVIEW

File: app/models/interest.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Interest < ActiveRecord::Base
 
   has_many :organization_interests, dependent: :destroy
-  has_many :organizations, through: :organization_interest
+  has_many :organizations, through: :organization_interests
```
Comment (Senen) [inline app/models/interest.rb]: Nice catch!

File: app/assets/javascripts/statistics.js.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+$(function(){
```
Comment (Senen) [inline app/assets/javascripts/statistics.js.erb]: This file has no ruby inside so there is no need to have **.erb** extension.

File: app/helpers/statistics_helper.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module StatisticsHelper
+  def self_employed(value)
+    hash_employed = Organization.lobby.all.map { |i| [i.name, i.represented_entities.count] }.to_h
```
Comment (Senen) [inline app/helpers/statistics_helper.rb]: This line should be:
```Organization.lobby.map { |i| [i.name, i.represented_entities.count] }.to_h```

File: app/views/statistics/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  <table class="display" cellpadding="0" width="100%" >
+    <thead>
+      <tr>
+        <th width="40%"><%= t('statistics.category')%></th>
+        <th width="15%"><%= t('statistics.amount')%></th>
+      </tr>
+    </thead>
+    <tbody>
+      <%@lobby_category.each do|name , number| %>
+      <tr>
```
Comment (Senen) [inline app/views/statistics/index.html.erb]: Review indentation and spacing!

File: config/initializers/assets.rb [hunk 5]
```diff
@@ -10,16 +10,17 @@
...
-Rails.application.config.assets.precompile += %w( ie_lt9.js )
-Rails.application.config.assets.precompile += %w( search_form.js )
-Rails.application.config.assets.precompile += %w( areas.js )
-Rails.application.config.assets.precompile += %w( events.js )
-Rails.application.config.assets.precompile += %w( holders.js )
-Rails.application.config.assets.precompile += %w( users.js )
-Rails.application.config.assets.precompile += %w( organizations.js )
-Rails.application.config.assets.precompile += %w( admin.css )
-Rails.application.config.assets.precompile += %w( email.css )
+Rails.application.config.assets.precompile += ['dependencies/html5shiv.min.js']
```
Comment (Senen) [inline config/initializers/assets.rb]: I do not understand this change!

File: config/routes.rb [hunk 6]
```diff
@@ -12,11 +12,10 @@
   get '/faq', to: 'questions#index', as: 'faq'
   get '/websitemap', to: 'static_pages#websitemap', as: 'websitemap'
 
-  resources :infringement_emails, only: [:new, :create]
+  resources :statistics, only: [:index]
```
Comment (Senen) [inline config/routes.rb]: Did we remove infringement emails?

File: app/views/statistics/index.html.erb (hunk 4 above)
Comment (Senen) [reply app/views/statistics/index.html.erb]: Make proper usage of Zurb Foundation accordion components. It should be initialized by default!

File: spec/features/visitors/statistics.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+feature 'Statistics page' do
+
+  scenario 'view statistics page' do
+    visit statistics_path
```
Comment (Senen) [inline spec/features/visitors/statistics.rb]: 1º We need to create some sample data to be able to check something inside each accordion, for now, this spec is not checking calculations.
2º Given When Then (I missed some empty lines)

File: app/views/statistics/index.html.erb (hunk 4 above)
Comment (vicentemendozam) [reply app/views/statistics/index.html.erb]: I changed the interface by the indicated one

File: config/initializers/assets.rb (hunk 5 above)
Comment (vicentemendozam) [reply config/initializers/assets.rb]: To add the new js, but now I deleted the file

**Comment by Senen** [comment]
Sorry @vicentemendozam as i said before at PR review comment, @alessandro is already fixing this! Sorry for the inconvenience!

**Comment by agileontheweb** [comment]
@Senen i pushed  this a new branch in new-statistics-page-refactor.
https://github.com/AyuntamientoMadrid/agendas/tree/new-statistics-page-refactor

Inside you can find this branch #292 with a  refactor.
After that vicentemendozam has inserted new changes , i merge this branch in new-statistics-page-refactor and solve conflicts.
I did not put new-statistics-page-refactor in pull-request, if you want i do this.

**Comment by Senen** [comment]
@agileontheweb please make a PR of you refactored branch and i make a review!

**Comment by Senen** [comment]
Closed and contained in favor of #298

**Pull request closed without merging**